Start enabling better use of webpack and module system

tomratcliffe opened the pull request.

tomratcliffe committed:
Remove font-awesome from source control

tomratcliffe committed:
Remove expanded.html and functionality

tomratcliffe committed:
Add jquery and webpack loader dependencies

tomratcliffe committed:
Remove jQuery libs from source control

tomratcliffe committed:
Remove built javascript files

tomratcliffe committed:
Update build config and remove expanded from build command

tomratcliffe committed:
Move genome files to JSON

tomratcliffe committed:
Update index, use npm libraries and update styling import

tomratcliffe committed:
Add module specific logic to main Genoverse file

MindaugasDukauskas commented on .gitignore:
```diff
@@ -1,3 +1,4 @@
 /data*
 *.DS_Store
 node_modules/*
+build
```
We should remove this line as we will need builded version

MindaugasDukauskas commented on js/Genoverse.js:
```diff
@@ -109,10 +114,14 @@ var Genoverse = Base.extend({
-          browser.loadedPlugins[plugin] = browser.loadedPlugins[plugin] || 'script';
-          deferred.resolve(arg);
+        if(runningInModule || true) {
+          return done();
```
You need to check 
On line 137
`$.getScript(js, getCSS);`
As it tries to pull scripts when in storybook

MindaugasDukauskas commented on js/Genoverse.js:
```diff
@@ -109,10 +114,14 @@ var Genoverse = Base.extend({
-        function done() {
-          browser.loadedPlugins[plugin] = browser.loadedPlugins[plugin] || 'script';
-          deferred.resolve(arg);
+        if(runningInModule || true) {
```
Shouldnt it be only `if(runningInModule) {`

MindaugasDukauskas commented on index.js:
```diff
@@ -81,3 +86,11 @@ require(__dirname + '/js/plugins/karyotype.js');
+require(__dirname + '/css/genoverse.css');
+
+// Plugins:
+require('./plugins.index.js')
```
So if you load this file. Lines above it with plugins is not needed anymore?

MindaugasDukauskas commented on webpack.config.js:
```diff
@@ -1,22 +1,56 @@
-  output  : { filename: 'genoverse.min.js', path: __dirname + '/js' },
+  output  : {
+    filename: '[name].min.js',
+    path: __dirname + '/build',
```
I had to add these two lines:
```
library: 'Genoverse',
libraryTarget: 'commonjs2'
```
As for some reason package doesn't export genoverse without it

MindaugasDukauskas reviewed (changes requested):
Left a few comments. Apart from that looks good

tomratcliffe committed:
Add correct library export webpack config

tomratcliffe committed:
Make correct logic checks for loading plugins

tomratcliffe committed:
Remove duplicated plugins import

tomratcliffe committed:
Add build folder back into source control

For now, the easiest way to build and then install this in another project will be to point to the specific commit or branch that we want to use.

This will need to include the built code, until we add CI processes to automatically build and tag the repo at certain points, or publish to NPM or similar

tomratcliffe commented:
Thanks @MindaugasDukauskas, I addressed the comments, could you have another look please?

MindaugasDukauskas commented on js/Genoverse.js:
```diff
@@ -133,7 +134,7 @@ var Genoverse = Base.extend({
       if (browser.loadedPlugins[plugin] || $('script[src="' + js + '"]').length) {
         getCSS();
-      } else {
+      } else if (!runningInModule) {
```
You still need to add `done()`

MindaugasDukauskas replied:
Could be it works like that. But I think done resolves `deffered`. Could be wrong

tomratcliffe replied:
I checked - it still works correctly in storybook, it doesn't try to re-load the plugins

MindaugasDukauskas replied:
👍

MindaugasDukauskas reviewed (approved):
Added one comment. Not sure if its valid. But looks good
